### PR TITLE
Adding per-cluster setting via ClusterProxyConfig CRD

### PR DESCRIPTION
**Pull request by crobby**

### CODE REVIEW

**Review by pmatseykanets — APPROVED** [review]
Looks good other than `Insecure` option, which I'm not really found of.

File: pkg/crds/yaml/generated/management.cattle.io_clusterproxyconfigs.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+          tokens.
+        properties:
+          apiVersion:
+            description: 'APIVersion defines the versioned schema of this representation
+              of an object. Servers should convert recognized schemas to the latest
+              internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources'
+            type: string
+          enabled:
+            description: Enabled indicates whether downstream proxy requests for service
+              account tokens is enabled. Enabled
```
Comment (pmatseykanets) [inline pkg/crds/yaml/generated/management.cattle.io_clusterproxyconfigs.yaml]: I suppose this was a leftover.

```suggestion
              account tokens is enabled.
```

File: pkg/crds/names.go
Comment (pmatseykanets) [inline pkg/crds/names.go:1]: We probably need to make sure these new resources are being backed up and restored properly. Possibly not in this repo.

**Review by pmatseykanets — APPROVED** [review]

File: pkg/crds/yaml/generated/management.cattle.io_clusterproxyconfigs.yaml (hunk 1 above)
Comment (crobby) [reply pkg/crds/yaml/generated/management.cattle.io_clusterproxyconfigs.yaml]: I removed the extra Enabled and regenerated

**Pull request merged**